### ISSUE TITLE
dollar and cent: variation in horizontal advance

### ISSUE THREAD

**Issue by MariannaPaszkowska** [googlefonts/googlesans-flex#811]
FB reports that `dollar` and `cent` vary across the GRAD and/or ROND in horizontal advance

```
Ensure VFs with duplexed axes do not vary horizontal advance.
Check ID: com.google.fonts/check/varfont/duplexed_axis_reflow
🔥 GoogleSansFlex[GRAD,ROND,opsz,slnt,wdth,wght].ttf
🔥 FAIL
The following glyphs have variation in horizontal advance due to duplexed axes (GRAD or ROND): dollar and cent [code: duplexed-causes-reflow]
```

Tested data: https://github.com/googlefonts/googlesans-flex/actions/runs/7670878627

**Comment by MariannaPaszkowska** [comment]
related #824

**Comment by MariannaPaszkowska** [comment]
still present in #825

**Comment by schriftgestalt** [comment]
It would be good if the test would print the axis location of the fail.

**Comment by octaviopardo** [comment]
this is driving me nuts, and makes me wonder if there is some additional issue somewhere else. In a 500+ glyphs font, where I create 108 masters automatically everytime I push, we keep getting the same error ONLY in these glyphs. And last G3 update thereothically (and I would say practically as I ve reviewed them myself) addresses the issue in brace layers.

**Comment by mekkablue** [comment]
I suspect a small delta (<0.5) caused by rounding in the brace layers. Glyphs has delta optimization built in, meaning that deltas with less than half a unit are dropped by default. I assume glyphsLib does not do the same, but I am guessing.

Quick fix: a post-production solution looking for, and removing, advance width deltas on the `GRAD` and `ROND` axes, in `gvar` and `HVAR` tables.

**Comment by MariannaPaszkowska** [comment]
@octaviopardo @mekkablue 
I found the issue in cent, searching in the dollar.
Have a look:
<img width="1013" alt="Screenshot 2024-02-08 at 3 07 26 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/938d40ef-906f-4d0b-b12e-76281b3ba76a">

<img width="1006" alt="Screenshot 2024-02-08 at 3 07 31 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/7a533120-8f79-448a-ae3a-49e9f807ae8f">

**Comment by MariannaPaszkowska** [comment]
It looks like something is going on in the heavy condensed corner there, and on the GRAD axis.

<img width="1006" alt="Screenshot 2024-02-08 at 3 09 44 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/4b0fbb9b-61ba-49de-b350-9a0bed702814">

<img width="1010" alt="Screenshot 2024-02-08 at 3 11 17 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/331c7573-ff62-47db-a3e8-9bb5df3f745c">

**Comment by MariannaPaszkowska** [comment]
ROND is affected too. I tested the import from yesterday: https://github.com/googlefonts/googlesans-flex/actions/runs/7819252075 
<img width="1008" alt="Screenshot 2024-02-08 at 3 13 34 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/60227c8d-00c0-48df-b3ad-b60102129d63">

<img width="1010" alt="Screenshot 2024-02-08 at 3 13 39 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/ac6eefff-9212-4ca0-ab93-6b29730f7a61">

**Comment by MariannaPaszkowska** [comment]
@octaviopardo I found something in the cent, in Italic, might be only spacing:
<img width="528" alt="Screenshot 2024-02-08 at 5 35 43 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/b59e58e7-aa43-4ca3-ad94-24bfb7bc9c40">
<img width="515" alt="Screenshot 2024-02-08 at 5 35 56 PM" src="https://github.com/googlefonts/googlesans-flex/assets/22956348/414293dd-4486-43d8-a2d1-d3d9b81f8216">

**Comment by RickyDaMa** [comment]
This issue has unfortunately crept back because of the merge of #885 - dependabot silently moved us back onto mainline glyphsLib during the dependency upgrade (97c2d99da206e2e24010c01bfbdb26a9c9f934f8). Not ideal behaviour, however in fairness to it we hadn't updated the `requirements.in` to use the hotfix branch (under the impression it'd reach mainline glyphs sooner rather than later)

**This means that v2.001 has the problem highlighted in this issue** (I've checked, it's there)

This will be fixed by #902 fortunately as that moves glyphsLib back onto the hotfix branch (both in `requirements.in` & `requirements.txt`)

![image](https://github.com/googlefonts/googlesans-flex/assets/113340856/2b27e97e-6727-4095-92c9-1eb9065df609)

Regression:
![image](https://github.com/googlefonts/googlesans-flex/assets/113340856/4c3d7f9e-0a84-49e6-a3b2-b2dd124837bc)

**Comment by belluzj** [comment]
Update on this issue: thankfully there's not been any regression in v2.001 compared to v2.000. Despite the glyphsLib dependency reverting to a version with the width bug, the production sources (UFOs) haven't been re-imported between v2.000 and v2.001, so the fonts behave the same.

The GIF below shows the issue (right-hand side of /dollar crashing badly into a vertical bar along the weight axis) in Beta4 of v2.000, and then shows that the issue is fixed in all of v2.000, v2.001, and the fonts built from the branch [dependency-updates](https://github.com/googlefonts/googlesans-flex/pull/902)

![gsflex_dollar_width_bug](https://github.com/googlefonts/googlesans-flex/assets/3616772/a351a438-ef7a-4106-8fb9-c8a55d49fcc5)


There are however a couple design issues around /dollar and /cent that I will report separately, but that are not related to the glyphsLib bugfix => reported here: https://github.com/googlefonts/googlesans-flex/issues/936
